src/ViewModel/CallToAction.php

```diff
@@ -13,19 +14,32 @@ final class CallToAction implements ViewModel
     use ArrayFromProperties;
 
     private $button;
+    private $dismissible;
     private $id;
     private $image;
     private $needsJs;
     private $text;
 
-    public function __construct(string $id, Picture $image, string $text, Button $button, bool $needsJs = false)
-    {
+    public function __construct(
+        string $id,
+        Picture $image,
+        string $text,
+        Button $button,
+        bool $needsJs = false,
+        DateTimeImmutable $cookieExpires = null
+    ) {
         Assertion::notBlank($id);
         Assertion::notBlank($text);
 
         $this->id = $id;
         if ($needsJs) {
             $this->needsJs = $needsJs;
+
+            if (null !== $cookieExpires) {
+                $this->dismissible = [
+                    'cookieExpires' => $cookieExpires->format(DATE_COOKIE),
```

Comment:
a language with :battery::battery: included

Reply:
Though produces an unexpected result for me
```
$ php -r "var_dump((new DateTimeImmutable())->format(DATE_COOKIE));"
string(34) "Thursday, 16-Jan-2020 13:52:54 UTC"
```
Uses `-` rather than spaces, but https://tools.ietf.org/html/rfc6265#section-5.1.1 seems to say that almost anything is allowed as delimiter e.g. `+` or `?`. In short, it works. :man_facepalming: